share/wake/lib/system/job.wake

```diff
@@ -919,7 +923,14 @@ export def defaultRunner: Runner =
     require Some _ = getenv "WAKE_EXPERIMENTAL_JOB_CACHE"
     else fuseRunner
 
-    mkJobCacheRunner fuseRunner
+    # The fuseRunner does not actully mount over / and instead uses the
+    # the host root as the sandbox root. This means that wakeroot will
+    # change from depending on where wake is being run from. As a hack
+    # to work around this we abuse the fact that the fuseRunner only
+    # works in relative paths to make different runs consistent. Ideally
+    # you'd have a more complex sandbox that kept the wakeroot at a
+    # consistent place across runs.
+    mkJobCacheRunner "/workspace" fuseRunner
```

Comment:
@mmjconolly I think after our discussion this is perhaps the main source of confusion since 1) I had this wrong and only working accidentally before but 2) I think it by happen stance works out to be our best option still. With the edits made in this change and this comment added for the one remaining super confusing issue. Does this make more sense?